packages/modeling/src/colors/colorize.js

```diff
@@ -52,10 +52,10 @@ const colorize = (color, ...objects) => {
   if (objects.length === 0) throw new Error('wrong number of arguments')
 
   const results = objects.map((object) => {
-    if (geom2.isA(object)) return colorGeom2(color, object)
-    if (geom3.isA(object)) return colorGeom3(color, object)
-    if (path2.isA(object)) return colorPath2(color, object)
-    if (poly3.isA(object)) return colorPoly3(color, object)
+    // if (geom2.isA(object)) return colorGeom2(color, object)
+    // if (geom3.isA(object)) return colorGeom3(color, object)
+    // if (path2.isA(object)) return colorPath2(color, object)
+    // if (poly3.isA(object)) return colorPoly3(color, object)
 
     object.color = color
```

Comment:
colorize should also return a shallow copy with new color applied. It is relatively ok to mutate for the color change as other operations do shallow copy, but then no other method should do mutation.

I would be ok this staying this way for this PR, but will likely push for change in a future PR to have consistency of immutability.

Reply:
Thanks for the review.

I'm hoping that geometry.clone() can become a shallow copy now, and will test this change. If a shallow clone() is workable then I'll update colorize() to use clone() again.